recipes/LJSpeech/quantization/train.py

```diff
@@ -1,142 +1,268 @@
+            self.hparams.train_logger.log_stats(
+                stats_meta={"Epoch loaded": current_epoch},
+                test_stats=stage_stats,
+            )
 
 
 def dataio_prepare(hparams):
+    """This function prepares the datasets to be used in the brain class.
+    It also defines the data processing pipeline through user-defined functions.
+
+    """
+    train_data = sb.dataio.dataset.DynamicItemDataset.from_json(
+        json_path=hparams["train_json"],
+        replacements={"DATA_ROOT": hparams["data_folder"]},
+    )
+    # For some reason the duration is not available in the manifest file...
+    # Sort training data to speed up training
+    # train_data = train_data.filtered_sorted(
+    #   sort_key="duration",
+    #    reverse=hparams["sorting"] == "descending",
+    #    key_max_value={"duration": hparams["train_remove_if_longer"]},
+    # )
+
+    valid_data = sb.dataio.dataset.DynamicItemDataset.from_json(
+        json_path=hparams["valid_json"],
+        replacements={"DATA_ROOT": hparams["data_folder"]},
+    )
+    # For some reason the duration is not available in the manifest file...
+    # Sort validation data to speed up validation
+    # valid_data = valid_data.filtered_sorted(
+    #    sort_key="duration",
+    #    reverse=True,
+    #    key_max_value={"duration": hparams["valid_remove_if_longer"]},
+    # )
+
+    test_data = sb.dataio.dataset.DynamicItemDataset.from_json(
+        json_path=hparams["test_json"],
+        replacements={"DATA_ROOT": hparams["data_folder"]},
+    )
+    # For some reason the duration is not available in the manifest file...
+    # Sort the test data to speed up testing
+    # test_data = test_data.filtered_sorted(
+    #   sort_key="duration",
+    #    reverse=True,
+    #    key_max_value={"duration": hparams["test_remove_if_longer"]},
+    # )
+
+    datasets = [train_data, valid_data, test_data]
+
+    # Define audio pipeline
+    takes = ["wav"]
+    provides = ["sig"]
 
-    # Define audio pipeline:
-    @sb.utils.data_pipeline.takes("wav")
-    @sb.utils.data_pipeline.provides("sig")
     def audio_pipeline(wav):
+        original_sample_rate = sb.dataio.dataio.read_audio_info(wav).sample_rate
         sig = sb.dataio.dataio.read_audio(wav)
-        info = torchaudio.info(wav)
-        resampled = torchaudio.transforms.Resample(
-            info.sample_rate,
-            hparams["sample_rate"],
-        )(sig)
-        return resampled
-
-    datasets = {}
-    data_info = {
-        "train": hparams["train_json"],
-    }
-    for dataset in hparams["splits"]:
-        datasets[dataset] = sb.dataio.dataset.DynamicItemDataset.from_json(
-            json_path=data_info[dataset],
-            replacements={"data_root": hparams["data_folder"]},
-            dynamic_items=[audio_pipeline],
-            output_keys=["id", "sig"],
+        sig = torchaudio.functional.resample(
+            sig, original_sample_rate, hparams["sample_rate"]
         )
+        yield sig
```

Comment:
why are you using this function with yield instead of the typical return?

Reply:
I typically use `yield` in pipelines so they can be readily extended to return multiple values if needed